### PR TITLE
Add new package ngspice-git.

### PR DESCRIPTION
**Pull request by stambaughw**
The current stable ngspice package has known issues that cause problems
with KiCad during simulations.  The issues have been resolved in the
master branch of ngspice.  This package allows you to build the latest
development version of ngspice with the bug fixes.  Of course there is
no guarantee that new bugs have not been introduced.  Use with caution.

### CODE REVIEW

**Comment by stambaughw** [comment]
I've added a new ngspice-git package for consideration.  This package allows you to create an ngspice package from the latest version of the ngspice git repo.  I did this because the current ngspice package has issues which prevent simulations from running correctly on KiCad.  This package solves these issues until the next stable release of ngspice.

Thanks,

Wayne

**Comment by Alexpux** [comment]
Don't see the reason to add this package. In last time I'm try to minimize git packages in repo. Even if I merge it I will not build it for repository. The better to solve issue is to apply patch for release sources.

**Comment by stambaughw** [comment]
Just for my knowledge.  What tool is being used to check the PKGBUILD  file quality so I can test any future packages before I submit them?

**Comment by mati865** [comment]
> Just for my knowledge. What tool is being used to check the PKGBUILD file quality so I can test any future packages before I submit them?

`saneman`

**Comment by stambaughw** [comment]
On 9/19/2016 11:35 AM, Mateusz Mikuła wrote:
>     Just for my knowledge. What tool is being used to check the PKGBUILD
>     file quality so I can test any future packages before I submit them?
> 
> |saneman|
> 
> —

There is no saneman package for either msys2 or mingw and I cannot find
the project source to try to build it for myself.  Is the only option to
create a pull request and wait for the results?

**Comment by mati865** [comment]
It is a part of `pactoys` package.

**Comment by stambaughw** [comment]
On 9/19/2016 11:54 AM, Mateusz Mikuła wrote:
> It is a part of |pactoys| package.

Got it!  Thanks!

File: mingw-w64-ngspice-git/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+pkgver=r4887.27053b8
+pkgrel=1
+pkgdesc='Mixed-level/Mixed-signal circuit simulator based on Spice3f5, Cider1b1, and Xspice.'
+url='http://ngspice.sourceforge.net'
+license=('BSD')
+arch=('any')
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
+             "git")
+depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")
+provides=("${MINGW_PACKAGE_PREFIX}-${_realname}"
```
Comment (Alexpux) [inline mingw-w64-ngspice-git/PKGBUILD]: No need to provide itself

File: mingw-w64-ngspice-git/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+package() {
+  cd "${srcdir}/build-static-${CARCH}"
+  make install DESTDIR="${pkgdir}"
+  install -D -m644 "${srcdir}/${_realname}/COPYING" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/LICENSE"
+  install -D -m644  "${srcdir}/${_realname}/doc/manual.pdf" "${pkgdir}${MINGW_PREFIX}/share/doc/${_realname}/manual.pdf"
+
+  # ngspice installs headers to /${PREFIX}/share/ngspice/include, which is not the usual location for *.h files
+  mv ${pkgdir}${MINGW_PREFIX}/share/ngspice/include ${pkgdir}${MINGW_PREFIX}
```
Comment (Alexpux) [inline mingw-w64-ngspice-git/PKGBUILD]: Maybe need patch build system to install in proper place? And send this patch upstream

**Comment by stambaughw** [comment]
On 10/13/2016 2:32 AM, Алексей wrote:
> *@Alexpux* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In mingw-w64-ngspice-git/PKGBUILD
> <https://github.com/Alexpux/MINGW-packages/pull/1720#pullrequestreview-4010600>:
> 
>> +#   Contributor: Luis Henrique <lmello.009@gmail.com>
> +#   Contributor: Wayne Stambaugh <stambaughw@gmail.com>
> +
> +_realname=ngspice-git
> +pkgbase=mingw-w64-${_realname}
> +pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
> +pkgver=r4887.27053b8
> +pkgrel=1
> +pkgdesc='Mixed-level/Mixed-signal circuit simulator based on Spice3f5, Cider1b1, and Xspice.'
> +url='http://ngspice.sourceforge.net'
> +license=('BSD')
> +arch=('any')
> +makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
> +             "git")
> +depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")
> +provides=("${MINGW_PACKAGE_PREFIX}-${_realname}"
> 
> No need to provide itself

This should be:

provides="${MINGW_PACKAGE_PREFIX}-ngspice"

since both ngspice and ngspice-git cannot be installed at the same time.

I'm curious as to why you've commented on this pull request since you
initially rejected it because you didn't want another -git package in
msys2.  If you are interested merging this then I will make the
appropriate changes.  If not, I'll just fix it in my repo.

**Comment by mingwandroid** [comment]
We are happy to merge it to this repo, just do not want to build -git binaries.

**Comment by stambaughw** [comment]
I appreciate that but I just don't have time to pick out the patches to
apply to the stable version of ngspice to make it work properly on
windows.  In order to build kicad, I had to either build and install the
latest git version from source or provide a pkgbuild.  The pkgbuild is
preferable to asking users to build ngspice from source.  Hopefully the
next stable version of ngspice will address the issues and the
ngspice-git package wont be necessary.

Thanks,

Wayne

**Comment by Alexpux** [comment]
Rebase changes on top of header

File: mingw-w64-ngspice-git/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+pkgrel=1
+pkgdesc='Mixed-level/Mixed-signal circuit simulator based on Spice3f5, Cider1b1, and Xspice. (mingw-w64)'
+url='http://ngspice.sourceforge.net'
+license=('BSD')
+arch=('any')
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
+             "git")
+depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")
+provides=("${MINGW_PACKAGE_PREFIX}-${_realname}"
+          "${MINGW_PACKAGE_PREFIX}-ngspice")
```
Comment (Alexpux) [inline mingw-w64-ngspice-git/PKGBUILD]: Provides contain two the same entries. Remove second.

**Comment by stambaughw** [comment]
Did the latest push resolve this issue?

**Comment by stambaughw** [comment]
Not sure what you mean by this?  I tried pushing just the changes but
git was not allowing me to this until I rebased against upstream.  What
git foo do I need to run in order to get this resolved.

**Comment by stambaughw** [comment]
Good catch.  Fixed.  I guess saneman isn't that sane after all. ;)

**Pull request merged**